cfgov/teachers_digital_platform/models/activity_index_page.py

```diff
@@ -0,0 +1,528 @@
+
+    def get_template(self, request):
+        template = 'teachers_digital_platform/activity_index_page.html'
+        if 'partial' in request.GET:
+            template = 'teachers_digital_platform/activity_search_facets_and_results.html'  # noqa: E501
+        return template
+
+    def dsl_search(self, request, *args, **kwargs):
+        """Search using Elasticsearch 7 and django-elasticsearch-dsl."""
+        all_facets = copy.copy(self.activity_setups.facet_setup)
+        card_setup = self.activity_setups.card_setup
+        total_activities = len(card_setup)
+        search_query = request.GET.get('q', '')
+        facet_dict = {"aggs": {}}
+        for facet in FACET_LIST:
+            facet_dict["aggs"].update({
+                "{}_terms".format(facet): {"terms": {"field": facet}}
+            })
+        dsl_search = ActivityPageDocument().search().from_dict(facet_dict)
+        if search_query:
+            terms = search_query.split()
+            for term in terms:
+                dsl_search = dsl_search.query(
+                    "bool",
+                    must=Q("multi_match", **{"query": term})
+                )
+            # Get facet counts to remove facets with no hits after a search
+            initial_search = dsl_search.execute()
+            facet_response = {facet: getattr(
+                initial_search.aggregations, f"{facet}_terms").buckets
+                for facet in FACET_LIST}
+            for facet, facet_config in FACET_MAP:
+                facet_ids = [item['key'] for item in facet_response[facet]]
+                is_nested = facet_config[1]
+                if is_nested:
+                    for parent in all_facets[facet]:
+                        for child in parent['children']:
+                            if child['id'] not in facet_ids:
+                                parent['children'].remove(child)
+                    if parent['id'] not in facet_ids:
+                        i = all_facets[facet].index(parent)
+                        del all_facets[facet][i]
+                else:
+                    for flat_facet in all_facets[facet]:
+                        if flat_facet['id'] not in facet_ids:
+                            all_facets[facet].remove(flat_facet)
+        else:
+            dsl_search = dsl_search.sort('-date')
+        selected_facets = {}
+        for facet, facet_config in FACET_MAP:
+            if facet in request.GET and request.GET.get(facet):
+                facet_ids = [
+                    int(value) for value in request.GET.getlist(facet)
+                    if value.isdigit()
+                ]
+                selected_facets[facet] = facet_ids
+                for entry in all_facets[facet]:
+                    if entry["id"] in facet_ids:
+                        entry["selected"] = True
+                        if entry.get("children"):  # a nested facet
+                            for child in entry["children"]:
```

Comment:
@higs4281 Wouldn't you need to check if the entry has a parent too? If the entry is selected and has a parent, the parent should either have an "indeterminate" or "checked" status too.

Reply:
This has been resolved.